docs/DafnyRef/Attributes.md

```diff
@@ -344,136 +343,217 @@ recursion was explicitly requested.
+
+To understand how to tweak verification,
+it is first useful to understand how Dafny verifies functions or methods.
+
+For every method, Dafny extracts _assertions_, as follows:
+
+* any explicit [`assert` statement](#sec-assert-statement) is _an assertion_.
+* A consecutive pair of lines in a [`calc` statement](#sec-calc-statement) forms _an assertion of their equality_.
+* Every function or method call with [`requires` clause](#sec-requires-clause) yields _one assertion per clause_
+  (special cases such as sequence indexing come with a special require clause that the index is within bounds).
+* Assignments `o.f := E;` yield an _assertion_ that `o.f` is allowed by the enclosing [`modifies` clause](#sec-loop-framing)
+* Assignments by predicate `x :| P(x)` yield an _assertion_ that `exists x :: P(x)`.
+* Every [`ensures` clause](#sec-ensures-clause) yields an _assertion_ at the end of the method.
+* ...
+
+It is useful to mentally visualize all these assertions as a list that roughly follows the order in the code[^complexity-path-encoding],
+except for `ensures` assertions that appear before in the code but, for verification purposes, would appear at the end.
+Thus, to prove or disprove a single assertion within this list, all the assumptions Dafny needs are 1) the global context,
+and 2) every preceding assumptions (and previous assertions transformed in assumptions).
+
+[^complexity-path-encoding]: All the complexities of the execution paths (if-then-else, loops, goto, break....) are, down the road and for verification purposes, cleverly encoded with variables recording the paths and guarding assumptions made on each path. In practice, a second clever encoding of variables enables grouping many assertions together, and recovers which assertion is failing based on the value of variables that the SMT solver returns.
+
+To achieve higher verification performance, Dafny collects all the assertions of one method into one single conjecture (an _assertion batch_) that it sends to the verifier, which tries to prove it correct:
+
+* If the verifier says it is correct[^smt-encoding], it means that all the assertions hold.
+* If the verifier returns a counter-example, this counter-example is used to determine both the failing assertion and the failing path.
+  Dafny will then query again the verifier with the assumption that that particular assert is valid, so that it can retrieve other failing assertions happening afterwards.[^caveat-about-assertion-and-assumption]
+* If the verifier returns `unknown` or times out, or even preemptively for difficult assertions or to reduce the chance that the verifier will ‘be confused’ by the many assertions in a large batch, Dafny partitions the assertions up into smaller batches[^smaller-batches]. An extreme case is the use of the `/oneAssertAtATime` command-line option, which causes Dafny to make one batch for each assertion.
```

Comment:
Have we renamed this option in the implementation? I like the name you use here better than `/vcsSplitOnEveryAssert`. :)

Reply:
Ahah indeed we hadn't. I'm adding this special case to Dafny so that it can handle it ! I like it better and easier to remember than `/vcsSplitOnEveryAssert`.